### ISSUE TITLE
All updates to an object field except for the last one are ignored

### ISSUE THREAD

**Issue by mstniy** [parse-community/Parse-SDK-JS#1450]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

When updating multiple nested fields under the same object field using `ParseObject.set`, all updates except for the last one are dropped.

### Steps to reproduce

```
const obj = new Parse.Object("MyObject");
obj.set("a",{});
await obj.save();
obj.set("a.b", 5);
obj.set("a.c", 6);
```

### Actual Outcome

`obj.attributes` should be `{a: {b:5, c:6}}`

### Expected Outcome

`obj.attributes` is `{a: {c:6}}`

###  Environment

Server
- Parse Server version: `irrelevant`
- Operating system: `Ubuntu 20.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local, irrelevant`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `5.0.6`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- Parse JS SDK version: `alpha (commit d084ac9be058d9b13f8ac2bbf04615f76dc2800d)`

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-alpha.7](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0-alpha.7)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0)
